cmd/controller-manager/app/options/options.go

```diff
@@ -39,33 +35,23 @@ type Options struct {
 
 // AddFlags adds flags to fs and binds them to options.
 func (o *Options) AddFlags(fs *pflag.FlagSet) {
-	fs.Var(flagutil.NewMapStringBool(&o.FeatureGates), "feature-gates", "A set of key=value pairs that describe feature gates for alpha/experimental features. "+
-		"Options are:\n"+strings.Join(utilfeature.DefaultFeatureGate.KnownFeatures(), "\n"))
-
 	fs.StringVar(&o.Config.FederationNamespace, "federation-namespace", util.DefaultFederationSystemNamespace, "The namespace the federation control plane is deployed in.")
-	fs.StringVar(&o.Config.ClusterNamespace, "registry-namespace", util.MulticlusterPublicNamespace, "The cluster registry namespace.")
-	fs.DurationVar(&o.Config.ClusterAvailableDelay, "cluster-available-delay", util.DefaultClusterAvailableDelay, "Time to wait before reconciling on a healthy cluster.")
-	fs.DurationVar(&o.Config.ClusterUnavailableDelay, "cluster-unavailable-delay", util.DefaultClusterUnavailableDelay, "Time to wait before giving up on an unhealthy cluster.")
+	o.setDefaults()
+}
+
+func (o *Options) setDefaults() {
+	o.FeatureGates = make(map[string]bool)
+	o.LimitedScope = false
+	o.ClusterMonitorPeriod = util.DefaultClusterMonitorPeriod
 
-	fs.BoolVar(&o.LimitedScope, "limited-scope", false, "Whether the federation namespace will be the only target for federation.")
-	fs.DurationVar(&o.ClusterMonitorPeriod, "cluster-monitor-period", time.Second*40, "How often to monitor the cluster health")
+	o.Config.ClusterNamespace = util.MulticlusterPublicNamespace
+	o.Config.ClusterAvailableDelay = util.DefaultClusterAvailableDelay
+	o.Config.ClusterUnavailableDelay = util.DefaultClusterUnavailableDelay
 
-	fs.DurationVar(&o.LeaderElection.LeaseDuration, "leader-elect-lease-duration", util.DefaultLeaderElectionLeaseDuration, ""+
-		"The duration that non-leader candidates will wait after observing a leadership "+
```

Comment:
(No action required) Would it make sense to retain the help text for removed parameters as comments on `FederationConfig`?

Reply:
I uses these messages in the chart README for related features. I guess it is a good way to add it in `FederationConfig` as comments.